service/gcsutils/remotefs/remotefs_os.go

```diff
@@ -239,7 +258,45 @@ func Mkfifo(in io.Reader, out io.Writer, args []string) error {
 	if err != nil {
 		return err
 	}
-	return unix.Mkfifo(args[0], uint32(perm))
+
+	if err := unix.Mkfifo(args[0], uint32(perm)); err != nil {
+		return err
+	}
+	return Sync()
+}
+
+// OpenFile works like os.OpenFile. Since the GCS process calling structure
+// does not enable us to keep state, OpenFile doesn't return a handle to the file.
```

Comment:
I'm not sure. I guess a workaround can be for docker to call `remotefs OpenFile` and then have a small protocol for reading/writing/seeking/closing through `stdin` & `stdout`